File: test-order-jupiter/src/test/java/com/jazzinjars/testorderjupiter/domain/RaceTest.java
package com.jazzinjars.testorderjupiter.domain;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class RaceTest {

    @Test
    void shouldCheckRace() {
        Race race = new Race(1, "Collserola");

        assertThat(race)
                .isNotNull()
                .satisfies(r -> {
                    assertThat(r.getId()).isEqualTo(1L);
                    assertThat(r.getName()).isEqualTo("Collserola");
                });
    }
}

File: test-order-jupiter/src/test/kotlin/com/jazzinjars/testorderjupiter/domain/RaceTest.kt
package com.jazzinjars.testorderjupiter.domain

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.util.*

class RaceTest {
    @Test
    fun `a race has an end date and participants`() {
        val endDate = Date()
        val participants = listOf(Participant(DNI("12345678L"), "Jon Viene"))
        val race = Race(endDate, participants)

        assertThat(race.endDate).isNotNull
        assertThat(race.participants).isNotEmpty
    }
}
